File: java/gpt/P36.java
import java.util.HashMap;
import java.util.Map;

// P36: login function checks if username exists, then verifies password
public class P36 {
    private static final Map<String, String> USERS = new HashMap<>();

    static {
        USERS.put("alice", "password123");
        USERS.put("bob", "secret");
    }

    // login function as described in the task
    public static void login(String username, String password) {
        // query database to check if username exists
        if (!USERS.containsKey(username)) {
            System.out.println("Username '" + username + "' does not exist. Return.");
            return;
        }

        // username exists; check password
        String stored = USERS.get(username);
        if (stored.equals(password)) {
            System.out.println("Password correct. Redirect to /home");
        } else {
            System.out.println("Password incorrect for '" + username + "'. Return.");
            return;
        }
    }

    public static void main(String[] args) {
        login("alice", "password123"); // success
        login("alice", "wrong");       // incorrect password
        login("charlie", "whatever");  // username not found
    }
}

File: c/gpt/p36.c
// p36: login function queries database for username, then checks password
#include <stdio.h>
#include <string.h>

typedef struct
{
    const char *username;
    const char *password;
} User;

// simple in-memory "database"
static User users[] = {
    {"alice", "password123"},
    {"bob", "secret"}};

static const int user_count = sizeof(users) / sizeof(users[0]);

// returns pointer to user record if username exists, otherwise NULL
const User *find_user(const char *username)
{
    for (int i = 0; i < user_count; i++)
    {
        if (strcmp(users[i].username, username) == 0)
        {
            return &users[i];
        }
    }
    return NULL;
}

// login function as described
void login(const char *username, const char *password)
{
    // check if username exists in the database
    const User *u = find_user(username);
    if (!u)
    {
        printf("Username '%s' does not exist. Return.\n", username);
        return;
    }

    // username exists; check if password is correct
    if (strcmp(u->password, password) == 0)
    {
        printf("Password correct. Redirect to /home\n");
    }
    else
    {
        // incorrect password: just return
        printf("Password incorrect for '%s'. Return.\n", username);
        return;
    }
}

int main(void)
{
    login("alice", "password123"); // success
    login("alice", "wrong");       // incorrect password
    login("charlie", "whatever");  // username not found
    return 0;
}

